### PR TITLE
(#4762) - Add switch to only build for node

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
Hrm, I remember being surprised I could pass arguments to npm run, possibly what broke 0.10 here, its a stable failure

**Comment by nolanlawson** [comment]
I think that was a feature added recently to `npm`. I'd be fine with dropping support for 0.10 BTW...

File: bin/build.js [hunk 1]
```diff
@@ -179,41 +183,45 @@ function cleanup() {
...
+}
+
+if (process.argv[2] === 'node') {
+  buildForNode();
+  return;
 }
 
-Promise.resolve().then(function () {
-  return Promise.all(['dist', 'lib', 'src_browser'].map(function (file) {
-    return rimraf(file);
```
Comment (nolanlawson) [inline bin/build.js]: Removing `rimraf()` here means that it's possible to accidentally commit unwanted files from the `dist ` directory

File: package.json [hunk 2]
```diff
@@ -96,10 +96,11 @@
   },
   "scripts": {
     "build": "node ./bin/build.js",
+    "build-node": "node ./bin/build.js node",
     "build-as-modular-es5": "sh bin/build-as-modular-es5.sh",
     "test-unit": "mocha tests/unit/test.extras.js && npm run build-as-modular-es5 && mocha tests/unit",
-    "test-node": "sh bin/test-node.sh",
-    "test-component": "mocha tests/component",
+    "test-node": "npm run build-node && sh bin/test-node.sh",
+    "test-component": "npm run build-node && mocha tests/component",
```
Comment (nolanlawson) [inline package.json]: So this solves the problem of contributors not having to type `npm run build` before `npm test`, but it does an unnecessary double build on Travis

**Comment by nolanlawson** [comment]
The double build is not such a big deal; I'm mostly just concerned about the `rimraf()` thing.  So I'd say let's fix the `rimraf()` and then :+1:.

File: bin/build.js [hunk 3]
```diff
@@ -129,7 +129,9 @@ function doRollup(entry, fileOut) {
 
 // build for Node (index.js)
 function buildForNode() {
-  return doRollup('src/index.js', 'lib/index.js');
+  return mkdirp('lib').then(function() {
+    doRollup('src/index.js', 'lib/index.js');
```
Comment (nolanlawson) [inline bin/build.js]: `return doRollup(...)`

**Comment by daleharvey** [comment]
Yeh really not worried about travis doing a double build, but I want to make sure we have an easy to figure out workflow for local dev, remembering to do builds in between test runs is pretty confusing. Added the rm -rf dist and good catch on the `return doRullup`

**Comment by nolanlawson** [comment]
Sorry, wasn't being clear - I'm also worried about `rm -rf lib`, because unnecessary files can end up in there as well (e.g. during `npm run build-as-modular-es5`).

Otherwise, looks good!

**Comment by daleharvey** [comment]
Ok cool, I added the `rimraf('lib')`, cheers - https://github.com/pouchdb/pouchdb/commit/443cd66992337093ec963a61d1d3d333d63c6a4b

**Pull request closed without merging**